packages/algob/test/stubs/algo-operator.ts

```diff
@@ -22,25 +23,44 @@ export class AlgoOperatorDryRunImpl implements AlgoOperator {
     throw new Error("Not implemented");
   }
 
-  getAssetByID (assetIndex: number | bigint): Promise<import("algosdk").AssetInfo> {
-    throw new Error("Not implemented");
+  getAssetByID (assetIndex: number | bigint): Promise<AssetInfo> {
+    if (assetIndex === 1) {
+      const res: AssetInfo = {
+        index: 1,
+        params: {
+          creator: "addr-1",
+          total: 1000,
+          decimals: 8,
+          'default-frozen': false,
+          'unit-name': "TKN",
+          name: "ASA-1",
+          url: "link",
+          'metadata-hash': "12312442142141241244444411111133",
+          manager: bobAcc.addr,
+          reserve: undefined,
+          freeze: bobAcc.addr,
+          clawback: undefined
```

Comment:
Let's add this `asaDef` in `/mock` and import here. That way we won't need to hardcode assertions in `checkTx` as well

Reply:
Yes good idea but we will have to hardcode because that's a Transaction object and this one is asadef.

Reply:
also we are not checking these values,
we check for values which are sent to network.

Reply:
I was thinking if we could do assertions for tx.fields based on their values fetched (either from network/custom params), instead of directly hardcoding them. Eg.
```js
import { mockAsaDef } from "/mock/..";
const fields = { manager: "", clawback: bobAcc.addr }; // used in test (custom fields to overwrite network values)

function checkTx(..) {
 ..
 assert.equal(tx.assetManager, mockAsaDef.manager); // assert with network field as this wasn't overwritten
 assert.equal(tx.assetClawback, fields.clawback); // assert with custom field as the network value should be overwritten
 ..
}
```

Reply:
actually we are using values we want to update and not what is already there.
Consider an example when we execute transaction we pass those values as asset fields. We are checking for those values, If you look at the code we are using values passed to network(they are not hardcoded) 
We need to match them with given values , and not the values that are in checkpoint

Reply:
will update accordingly